operator-framework-junit5/src/main/java/io/javaoperatorsdk/operator/junit/OperatorExtension.java

```diff
@@ -47,24 +50,42 @@
   private final KubernetesClient kubernetesClient;
   private final ConfigurationService configurationService;
   private final Operator operator;
+  private final List<HasMetadata> operatorDeployment;
+  private final Duration deploymentTimeout;
   private final List<ReconcilerSpec> reconcilers;
+  private final List<HasMetadata> infrastructure;
+  private final Duration infrastructureTimeout;
   private final boolean preserveNamespaceOnError;
   private final boolean waitForNamespaceDeletion;
+  private final boolean oneNamespacePerClass;
+  private final boolean localOperatorDeployment;
 
   private String namespace;
 
   private OperatorExtension(
       ConfigurationService configurationService,
       List<ReconcilerSpec> reconcilers,
+      List<HasMetadata> operatorDeployment,
+      Duration deploymentTimeout,
+      List<HasMetadata> infrastructure,
+      Duration infrastructureTimeout,
       boolean preserveNamespaceOnError,
-      boolean waitForNamespaceDeletion) {
+      boolean waitForNamespaceDeletion,
+      boolean oneNamespacePerClass,
+      boolean localOperatorDeployment) {
 
     this.kubernetesClient = new DefaultKubernetesClient();
     this.configurationService = configurationService;
     this.reconcilers = reconcilers;
+    this.operatorDeployment = operatorDeployment;
+    this.deploymentTimeout = deploymentTimeout;
+    this.infrastructure = infrastructure;
+    this.infrastructureTimeout = infrastructureTimeout;
     this.operator = new Operator(this.kubernetesClient, this.configurationService);
```

Comment:
maybe this should be passed in the constructor and created by the builder I don't think there is a case where you need both a local operator and a deployed operator